### DISCUSSION TITLE
Entry fields - ability to include / configure add new entry button on entries select modal

### DISCUSSION THREAD

**Discussion by deanfields** [craftcms/cms#11499] [Ideas] [answered]
I have created two sources, both set up with criteria to only show entries with a particular entry type from the same section.  I selected one of these as the source on my entry custom field.

However, when using an entry field like this, you no longer get the 'add new entry' button In the bottom left corner. It would be fantastic if you're able to configure the 'add new entry' button to show only the entry types you would like a user to be able to create from that modal. 

If I just set the section for the source and use 'selectable entries condition' (on the field) to only show one entry type, the 'Add new entry' button appears, however, it defaults to the first entry type for that section which you have to change. I want users to only be able to create a specific entry type at that point, or at the very least pre-select it when the slideout for editing appears.

I understand this would be quite complex to automate when a source is selected to determine what section types should be allowed to be created, which is why I think some control over this on the configuration of an entry type field would be amazing. That also allows you to designate entry types instead of sections for the button.

**Comment by brandonkelly** [comment, marked as answer]
I’ve just implemented this for Craft 4.5 (#13404). Any custom source that is limited to only one section will automatically show a “New [section] entry” button within entry selection modals.

If any entry types are included in the custom source’s condition, the new entry will be set to the first selected entry type by default as well.

The Entries index page will get a primary “New entry” button for custom sources as well. (#13390)

**Reply by davidhellmann** [reply]
Is that also possible within the listview?

**Reply by davidhellmann** [reply]
I read the next item in the changelog 😅 Nice!

**Reply by brandonkelly** [reply]
Craft 4.5.0 is out with this!
